source/extensions/stat_sinks/hystrix/hystrix.cc

```diff
@@ -304,6 +321,32 @@ void HystrixSink::flush(Stats::Source&) {
   incCounter();
   std::stringstream ss;
   Upstream::ClusterManager::ClusterInfoMap clusters = server_.clusterManager().clusters();
+
+  // Save a map of the relevant histograms per cluster in a convenient format.
+  std::unordered_map<absl::string_view, std::unordered_map<std::string, double>, StringViewHash>
+      time_histograms;
+  for (const Stats::ParentHistogramSharedPtr histogram : server_.stats().histograms()) {
+    // histogram->name() on clusters of the format "cluster.cluster_name.histogram_name"
+    // i.e. "cluster.service1.upstream_rq_time".
+    const std::vector<absl::string_view> split_name = absl::StrSplit(histogram->name(), '.');
+    if (split_name[0] == "cluster" && split_name[2] == "upstream_rq_time") {
```

Comment:
check split_name.size()>2 first

Reply:
Do you mean there could be a case where the histogram name is made up of more parts?

Reply:
more parts would be OK. less parts would crash.  I would prefer to be locally paranoid and have an assertion rather than a SEGV if that occurs.